classes/Model/AnnoucementManagement.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * AnnoucementManagement Model
+ *
+ * @package TheyWorkForYou
+ */
+
+namespace MySociety\TheyWorkForYou\Model;
+
+function select_based_on_weight($items)
+{
+    # banners have a weight attribute, which is the probability of being selected
+    # the higher the weight, the higher the probability
+    $total_weight = 0;
+    foreach ($items as $item) {
+        $total_weight += $item->weight;
+    }
+
+    $random_number = rand(0, $total_weight);
```

Comment:
If I've understood this function correctly, the 0 should be 1? As otherwise the first item in the list will have an extra 0 slot to get picked (e.g. three items, each of weight 1, rand will be 0, 1, 2, or 3, and first item will be 50%, others 25%).